beacon_node/beacon_chain/src/data_availability_checker/overflow_lru_cache.rs

```diff
@@ -331,27 +355,38 @@ impl<T: BeaconChainTypes> OverflowLRUCache<T> {
     }
 
     pub fn has_block(&self, block_root: &Hash256) -> bool {
-        self.critical.read().has_block(block_root)
+        let read_lock = self.critical.read();
+        if read_lock
+            .in_memory
+            .peek(block_root)
+            .map_or(false, |cache| cache.executed_block.is_some())
+        {
+            true
+        } else if read_lock.store_keys.contains(block_root) {
+            drop(read_lock);
+            // I assume if there's some kind of error reading from the store, we should just return false
```

Comment:
Seems reasonable to me
```suggestion
            // If there's some kind of error reading from the store, we should just return false
```